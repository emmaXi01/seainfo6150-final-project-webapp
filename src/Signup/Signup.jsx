import React, { useState } from 'react';
import PropTypes from "prop-types";
import { addUser } from "../service.js";
import styles from "./Signup.module.css";

const Signup = (props) => {
  const[newUsername, setNewUsername] = useState('');
  const[newPassword, setNewPassword] = useState('');
  const [errorText, setErrorText] = useState('');

  const onUsernameChange = (e) => {
    setNewUsername(e.target.value);
  };

  const onPasswordChange = (e) => {
    setNewPassword(e.target.value);
  };

  const performSignup = () => {
    if(!newUsername || newUsername.trim() === "") {
      setErrorText("You should Enter an User Name!");
      return;
    }

    if(!newPassword || newPassword.trim() === "") {
      setErrorText("You should Enter a Password!");
      return;
    }

    const user = {"name": newUsername, "password": newPassword};
    addUser(user);

    setErrorText('');
    props.onSignup(newUsername);
  };

  return(
    <div className={styles.container}>
      <h4 className={styles.greeting}>Sign Up and Start Learning!</h4>
      <p className="error">{errorText}</p>
      <div className="signup-body">
        <input className={styles.username} onChange={onUsernameChange} value={newUsername} placeholder="User name"/>
        <input className={styles.password} onChange={onPasswordChange} value={newPassword} placeholder="Password"/>
        <button className={styles.sigupButton} onClick={performSignup }>Sign Up</button>    
      </div>     
    </div>
  );
};

Signup.propTypes = {
  onSignup: PropTypes.func.isRequired
};

export default Signup;