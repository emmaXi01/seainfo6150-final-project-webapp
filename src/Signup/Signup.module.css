.container {
  width: 500px;
  margin: 5% auto 5%;
  padding: 50px 80px;
  border: 1px solid #ccc;
  border-radius: 10px;
}

.greeting {
  display: block;
  text-align: center;
  color: #29303b;
  font-weight: 700;
  font-size: 20px;
  padding-bottom: 10px;
}

.username, .password {
  width: 100%;
  padding: 12px 10px;
  font-size: medium;
  margin: 8px 0; 
  border: 1px solid #ccc;
  box-sizing: border-box;
}

.sigupButton {
  background-color: #4085d3;
  color: white;
  font-size: medium;
  padding: 14px 10px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 100%;
} 

@media screen and (min-width: 400px) {
  .container{
    width: 300px;
  }
} 

@media screen and (min-width: 600px) {
  .container{
    width: 400px;
  }
}

@media screen and (min-width: 1150px) {
  .container{
    width: 600px;
  }
}