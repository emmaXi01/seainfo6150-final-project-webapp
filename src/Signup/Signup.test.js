import React from "react";
import { render } from "@testing-library/react";
import Signup from "./Signup.jsx";

describe("Signup tests", () => {
  it("renders the signup component correctly", () => {
    const { container } = render(<Signup onSignup={()=>{}}/>);
    expect(container).toMatchSnapshot();
  });
});