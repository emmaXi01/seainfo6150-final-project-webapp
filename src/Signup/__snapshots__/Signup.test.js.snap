// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup tests renders the signup component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h4
      class="greeting"
    >
      Sign Up and Start Learning!
    </h4>
    <p
      class="error"
    />
    <div
      class="signup-body"
    >
      <input
        class="username"
        placeholder="User name"
        value=""
      />
      <input
        class="password"
        placeholder="Password"
        value=""
      />
      <button
        class="sigupButton"
      >
        Sign Up
      </button>
    </div>
  </div>
</div>
`;
